docs/custom-components/custom-components-command-reference.md

```diff
@@ -0,0 +1,202 @@
+
+```python eval
+doccmdoutput(
+    command="reflex component --help",
+    output="""Usage: reflex component [OPTIONS] COMMAND [ARGS]...
+
+  Subcommands for creating and publishing Custom Components.
+
+Options:
+  --help  Show this message and exit.
+
+Commands:
+  build    Build a custom component.
+  init     Initialize a custom component.
+  publish  Publish a custom component.
+""")
+```
+
+## reflex component init
+
+Below is an example of running the init command.
+
+```python eval
+doccmdoutput(
+    command="reflex component init",
+    output="""reflex component init
+Info: Component class name: GoogleAuth
+Info: Package name: reflex-google-auth
+Info: Custom component source directory: reflex_google_auth
+Info: Demo app directory: google_auth_demo
+Info: Populating pyproject.toml with package name: reflex-google-auth
+Info: Initializing the component directory: custom_components/reflex_google_auth
+Info: Creating app for testing: google_auth_demo
+──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── Initializing google_auth_demo ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+[12:07:32] Initializing the app directory.
+Success: Initialized google_auth_demo
+Info: Installing reflex-google-auth in editable mode.
+Info: Package reflex-google-auth installed!
+Custom component initialized successfully!
+Here's the summary:
+pyproject.toml and README.md created. Please fill in details such as your name, email, homepage URL.
+Source code template is in custom_components. Start by editing it with your component implementation.
+Demo app created in google_auth_demo. Use this app to test your custom component.
+""",
+)
+```
+
+The init command uses the current enclosing folder name to construct a python package name, typically in the kebab case. For example, if running init in folder `google_auth`, the package name will be `reflex-google-auth`. The added prefix reduces the chance of name collision on PyPI (the Python Package Index), and it indicates that the package is a Reflex custom component. The user can override the package name by providing the `--package-name` option.
+
+The init command creates a set of files and folders prefilled with the package name and other details. During the init, the `custom_component` folder is installed locally in editable mode, so a developer can incrementally develop and test with ease. The changes in component implementation is automatically reflected where it is used. Below is the folder structure after the init command.
+
+```python eval
+rx.code_block("""
+google_auth/
+├── pyproject.toml
+├── README.md
+├── custom_components/
```

Comment:
Not related to this PR, but why can't we remove the entire `custom_components` layer, and move the `reflex_google_auth` folder up to the root directory? Is it if they want to have multiple custom components? But then that doesn't really fit with our whole demo structure anyways. So maybe for now we just say 1 repo = 1 custom component?

Reply:
What are we trying to achieve removing the "custom_components" directory. Check the python package guide: https://packaging.python.org/en/latest/tutorials/packaging-projects We're not adding anything extra, except we call it "custom_components" instead of "src". As I mentioned, the "custom_componnets" is where to find the source, "custom components" is not included. When you install the package, what's inside it gets installed, e.g. `reflex_spline`, inside that, there init and the implementation file.

Reply:
Ah got it, didn't realize it needed that extra directory in the guide